src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -46,80 +77,1172 @@ const ActorInit En_Kakasi_InitVars = {
+}
+
+extern FlexSkeletonHeader D_060065B0;
+
+void EnKakasi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi* this = (EnKakasi*)thisx;
+    s32 tempCutscene;
+    s32 i;
+
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &D_80971D80);
+    SkelAnime_InitSV(globalCtx, &this->skelanime, &D_060065B0, &D_06000214, 0, 0, 0);
+
+    this->songSummonDist = GET_KAKASI_SUMMON_DISTANCE(this) * 20.0f;
+    if (this->songSummonDist < 40.0f) {
+        this->songSummonDist = 40.0f;
+    }
+
+    this->unkHeight = (this->actor.world.rot.z * 20.0f) + 60.0f;
+    this->actor.world.rot.z = 0;
+    this->actor.targetMode = 0;
+    if (this->actor.world.rot.x > 0 && this->actor.world.rot.x < 8) {
+        this->actor.targetMode = this->actor.world.rot.x - 1;
+    }
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+
+    this->aboveGroundStatus = GET_KAKASI_ABOVE_GROUND(this);
+    this->actor.world.rot.x = 0;
+    this->actor.flags |= 0x400;
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    Actor_SetScale(&this->actor, 0.01f);
+
+    i = 0;
+    tempCutscene = this->actor.cutscene;
+    while (tempCutscene != -1) {
+        // petrie911 has suggested this might be a fake match, no others found yet
+        tempCutscene = ActorCutscene_GetAdditionalCutscene(this->actorCutscenes[i] = tempCutscene);
+        i++;
+    }
+
+    if (this->aboveGroundStatus) {
+        if (gSaveContext.weekEventReg[79] & 0x8) {
+            this->aboveGroundStatus = ENKAKASI_ABOVE_GROUND_TYPE;
+            this->songSummonDist = 80.0f;
+            EnKakasi_SetupIdleUnderground(this);
+        } else {
+            Actor_SetHeight(&this->actor, 60.0f);
+            this->unkFunc = EnKakasi_8096F88C;
+            if (gSaveContext.weekEventReg[83] & 0x1) {
+                EnKakasi_InitTimeSkipDialogue(this);
+            } else {
+                EnKakasi_SetupIdleStanding(this);
+            }
+        }
+    } else {
+        EnKakasi_SetupIdleUnderground(this);
+    }
+}
+
+void EnKakasi_SetAnimation(EnKakasi* this, s32 index) {
+    this->animeIndex = index;
+    this->animeFrameCount = SkelAnime_GetFrameCount(&kakasiAnimations[this->animeIndex]->common);
+    // 1: regular playback speed, 0: starting frame
+    SkelAnime_ChangeAnim(&this->skelanime, kakasiAnimations[this->animeIndex], 1.0f, 0.0f, this->animeFrameCount,
+                         D_8097206C[this->animeIndex], -4.0f);
+}
+
+/*
+ * set by init, called... nowhere? left over from OOT?
+ */
+void EnKakasi_8096F88C(GlobalContext* globalCtx, EnKakasi* this) {
+    func_8013A530(globalCtx, &this->actor, 0x7, &this->actor.focus.pos, &this->actor.shape.rot, 280.0f, 1800.0f, -1);
+}
+
+void EnKakasi_CheckAnimateSFX(EnKakasi* this) {
+    if (this->animeIndex == ENKAKASI_ANIME_SIDEWAYS_SHAKING || this->animeIndex == ENKAKASI_ANIME_ARMS_CROSSED_STILL) {
+        if (func_801378B8(&this->skelanime, 1.0f) != 0 || func_801378B8(&this->skelanime, 8.0f) != 0) {
```

Comment:
```suggestion
        if (func_801378B8(&this->skelanime, 1.0f) || func_801378B8(&this->skelanime, 8.0f)) {
```
Worth double checking, but you might not need all the `!= 0` in this function